src/Nethermind/Nethermind.Sockets/ReceiveResult.cs

```diff
@@ -1,13 +1,20 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
-namespace Nethermind.Sockets
+namespace Nethermind.Sockets;
+
+public readonly struct ReceiveResult
 {
-    public class ReceiveResult
+    private readonly bool _isNotNull;
+
+    public readonly bool IsNull => !_isNotNull;
+
+    public ReceiveResult()
     {
-        public int Read { get; init; }
-        public bool EndOfMessage { get; init; }
-        public bool Closed { get; init; }
-        public string? CloseStatusDescription { get; init; }
+        _isNotNull = true;
```

Comment:
this is always true?

Reply:
Only when struct constructor is explicitly called, `default` value will have it at `false`.

Reply:
Would be good to document with comment

Reply:
Have a test for it